### ISSUE TITLE
[question] Conan telemetry / analytics data

### ISSUE THREAD

**Issue by ilya-lavrenov** [conan-io/web#128]
### What is your question?

Hi,

We have recently added OpenVINO to Conan Center Index and we track telemetry data via:
![image](https://github.com/conan-io/conan/assets/2566854/c604b47b-99b9-48ad-93dc-493a2cb50e16)

It's quite strange that some days a number of downloads is always 18. It makes me thinking that it's a data from some CI service (maybe conan CI? when we send some updates to existing conanfile.py and Conan CI rebuilds the recipe?). Is it true? do we need to count data from CI service? 

I suppose we need to count only downloads from real users.

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
Transferred to the web repo, as this is not something that belongs to the Conan client

**Comment by ilya-lavrenov** [comment]
> Transferred to the web repo, as this is not something that belongs to the Conan client

is it not a client who reports download statistic?
E.g. if client is run in CI service, it must not report the statistic.

**Comment by memsharded** [comment]
The client has not telemetry, statistics or any kind of data collection at all.
The numbers of those graphics are exclusively from ConanCenter servers statistics.

**Comment by ilya-lavrenov** [comment]
But only client knows about environment it's running in and it should provide this info to ConanCenter servers to track statistic correctly, is not it?

**Comment by memsharded** [comment]
Not really about the environment, but when a certain binary is requested from the server, the client sends the ``package_id`` and the summary of settings and options of that ``package_id``. It is not really a representation of the environment, because you can query or download packages for different profiles than the current configuration. That is part of the client-server Conan protocol, not the typical telemetry.

**Comment by ilya-lavrenov** [comment]
I meant real environment, e.g. almost all CI services define `CI` variable to `true`.
If Conan Server does not know information about it, it can count such download request to download statistic. But I suppose `conan` client should pass such information as well, so Conan Server can at least differentiate what certain binary is used within CI (github actions, travis, appveyor, etc) and count such statistic differently, otherwise if someone uses package in CI, such package will have a great number of downloads, while it's actually a single user.

**Comment by memsharded** [comment]
> If Conan Server does not know information about it, it can count such download request to download statistic. But I suppose conan client should pass such information as well, so Conan Server can at least differentiate what certain binary is used within CI (github actions, travis, appveyor, etc) 

We are certainly filtering out all traffic from ConanCenter CI itself, so ``conan-center-index`` builds should never affect.
Otherwise, the Conan HTTP API to the server is basically a "GET" request with the ``package_id`` and the other associated ``package_id`` information, there is absolutely no other telemetry data from the client, we cannot know if it is a CI service, a developer, a bot or elsewhere. And at this moment, we don't plan to add any other telemetry to the Conan client to gather information about their environment.

**Comment by ilya-lavrenov** [comment]
> we cannot know if it is a CI service, a developer, a bot or elsewhere

You can filter out at least cases when `CI` env var is set.
See https://github.com/orgs/community/discussions/26628#discussioncomment-3252605

**Comment by memsharded** [comment]
> You can filter out at least cases when CI env var is set.

As a side note, I am not so sure that CI builds shouldn't count downloads. I think that other central repos do not take into account if the downloads come from CI agents or developer machines. The fact that your library is being pulled by a CI agent doesn't mean that it is not a valuable download, but rather the opposite, it is being used in production almost for sure.

But in any case, it doesn't matter, the Conan client will not collect information about user system env-vars and send it to the ConanCenter server.

**Comment by ilya-lavrenov** [comment]
> do not take into account if the downloads come from CI agents or developer machines

But people want to differentiate such statistic and create tickets https://discuss.python.org/t/tracking-interactive-vs-ci-use-of-pip-and-thus-pypi-downloads/7179/7
Otherwise, product which is used in a single CI which has a lot of builds per day, can be as popular as one which is used by real users interactively. But interactive counts are more important 
At least, there should be a way to differentiate these downloads.

> not a valuable download, but rather the opposite, it is being used in production almost for sure.

But it case be used only once in production, while all other downloads are cheated because of CI. 

> the Conan client will not collect information about user system env-vars and send it to the ConanCenter server.

Why not? It can collect only `CI` env var, which is harmless.

**Comment by memsharded** [comment]
> But people want to differentiate such statistic and create tickets https://discuss.python.org/t/tracking-interactive-vs-ci-use-of-pip-and-thus-pypi-downloads/7179/7

Sure, people want to differentiate it. Still the tickets that I have been reading in the Python trackers and things like PyPIstats not publishing metrics for CI vs non-CI, more than 2.5 years after the last comment in the discussion that you posted is very revealing.

> Why not? It can collect only CI env var, which is harmless.

Nothing is harmless. After we add the ``CI`` variable, there will be requests to add another bunch of variables like CIRCLE_CI, JENKINS_CI, etc, etc. That is expensive support and maintenance cost. And then one day, one big enterprise realize that the env-var they are using called ``CI`` that they store some CI sensitive information is being sent by Conan to ConanCenter. Very unlikely, but if you know about Hyrum's Law, it will eventually happen. Or mandatory https://xkcd.com/1172. Yes, I know, they shouldn't do that in the first place, but when you have to wake up at 3am on a Saturday to handle this because the world is on fire because of this error, then deal with the the reputation impact too (because people like to viralize these things), etc, etc., makes you reconsider your choices. 

We could continue iterating, that ok, but you didn't need to send the value, just "cast" it to 1 or 0, then users checking the code, complaining that their "False" literal string is being casted to 1, then a new PR to change it (on top of the others for adding the other env-vars). And still there will be enterprise users upset that we are leaking information from their systems to our server, even if it is just a boolean. And then we could open the discussion for mirrors and bots, and keep going further down the rabbit hole.

So I am sorry, but no. We have too many scars because of things like this. It is not worth the effort and the risks, Conan client will not be sending system env-vars information to the server, and ConanCenter WebUI will show just an aggregated download count of all the downloads, irrespective of who/what/how is downloading it. In an ideal world with infinite resources, it would be possible, but we need to prioritize, and this is very far from being something that needs to be prioritized.

**Comment by ilya-lavrenov** [comment]
Hi @davidsanfal
Could you please explain why do we have such picks with 1 downloads per day?
How can I convert it to real users? I hardly can imagine that 18 people downloaded OpenVINO recipe x 3 days.

**Comment by davidsanfal** [comment]
Hi @ilya-lavrenov.

We have removed the statistics after finding some inconsistencies in the numbers that were being reported. We discovered that the totals were miscalculated, but the per-version recipe download counts were very under-reported (there were actually more downloads to report, at least for the few recipes that we sampled). Re-calculating the correct numbers involves processing large amounts of access log data. This is not currently a priority for the team as it involves additional resources.

On the other hand, we can confirm that the statistics _never_ included requests originating from the Conan Center CI service itself - this is by design. We can only report requests against the Conan Center remote, as the Conan client categorically does not have any telemetry. Due to this, we are unable to segregate use (we don't know and can't know if requests come from somebody else's CI service). For the library you have reported, once the correct statistics are calculated, the "peaks" that you saw. We routinely monitor for abuse/spam, but besides that we do not concern ourselves with the origin of the requests.

Additionally bear in mind that due to different layers of caching (conan local cache on users or CI machines, third-party Artifactory remotes, and forks of Conan Center outside of our view), it is difficult to ascertain the actual numbers of any library, as any of those methods are not visible to us. Thus the numbers we get are an estimate at best, as it is currently impossible for us to determine the actual usage of any recipe.

**Comment by ilya-lavrenov** [comment]
Thanks for your answer @davidsanfal 

Conan is not the only package manager who uses caches, etc (pip does it as well) and other still can provide some usage statistic.
I'm not asking about precise data, but it's better to have something than nothing. Based on download stat different users / companies can understand whether they should / should not invest time in support of specific packages. 

I see your plans, but anyway it would be very helpful.
